### PR TITLE
'BAG {} OF ENUM' support

### LINKED ISSUE #119: Attributes of type ENUMs:  Forms are not created correctly (no drop-down lists or similar)  because of missing relations or other reasons
If an attribute with ENUM type is defined in an abstract class, projectgenerator doesn't create a relation in QGIS although the ENUM table is marked as "ENUM" in the metadatatable t_ili2db_table_prop.

Therefore, the widget type in the form Definition is not defined as "Relation reference", but "Text Edit".

In the attached example the abstract class "Hecken_Feldgehoelze_Basis" (with several attributes of type "ENUM") is extended to the classes "Hecken_Feldgehoelze_Linie" and "Hecken_Feldgehoelze_Flaeche" where the QGIS relation definitions to the ENUM tables are missing.

[ZG_Hecken_und_Feldgehoelze_V1_0.txt](https://github.com/opengisch/projectgenerator/files/1556215/ZG_Hecken_und_Feldgehoelze_V1_0.txt)

### LINKED ISSUE #171: Bag of Enums
Should be displayed as lists where multiple entries can be selected/checked.
To decide: relation reference or value relation.

### PR DESCRIPTION
**Pull request by gacarrillor**
bags_of_enum

bags_of_enum_2

- [x] Tests for PostGIS.
- [x] Tests for GeoPackage.
- [x] Support for domains from Structures defined out of topics (Close #119)
- [x] Configure Value Relations for BAGs OF ENUM if `!!@ili2db.mapping=ARRAY` meta-attribute is found.

Fix #119
Fix #171

### CODE REVIEW

**Comment by gacarrillor** [comment]
@m-kuhn, do you know if constraints for Value Relations are working well? Perhaps I'm missing something, but passing 


```
{ 
  'AllowNull': False,   <----- Not sure if this works as expected...
  'AllowMulti': True, 
  'UseCompleter': False, 
  'Value': 'dispName', 
  'OrderByValue': True, 
  'Layer': 'a_layer_id', 
  'FilterExpression': '', 
  'Key': 'iliCode', 
  'NofColumns': 1
}
```


as configuration dict, doesn't force users to select at last one option in the checklist, that is, there is no orange cross alongside the field widget if there is no selection.

**Comment by m-kuhn** [comment]
AllowNull is (so far) not implmented for AllowMulti widgets. In fact, it's not even a NOT NULL but a NOT EMPTY (or similar) constraint in this situation.
See also https://github.com/opengisch/projectgenerator/issues/182

**Comment by gacarrillor** [comment]
@m-kuhn This PR is done, please give it a review. If you have any question just let me know.

File: projectgenerator/tests/test_projectgen_generic_databases.py [hunk 1]
```diff
@@ -104,7 +104,8 @@ def test_postgis_db_with_non_empty_and_no_interlis_schema_with_spatial_tables(se
             layers = generator.layers()
 
             self.assertEqual(len(layers), 2)
-            self.assertEqual(len(generator.relations(layers)), 1)
+            relations, foo = generator.relations(layers)
```
Comment (m-kuhn) [inline projectgenerator/tests/test_projectgen_generic_databases.py]: Better replace `foo` with `_`, then it's clear that it's unused.

**Comment by m-kuhn** [comment]
Nice, thanks a lot!

**Comment by m-kuhn** [comment]
Thanks a lot!!

**Pull request merged**